test/libyul/yulOptimizerTests/fullSuite/unusedFunctionParameterPruner_loop.yul

```diff
@@ -0,0 +1,33 @@
+{
+    sstore(f(1), 1)
+    sstore(f(2), 1)
+    sstore(f(3), 1)
+    function f(a) -> x {
+        for {let b := 10} iszero(b) { b := sub(b, 1) }
+        {
+            a := calldataload(0)
+            mstore(a, x)
+        }
+    }
+}
+// ----
+// step: fullSuite
+//
+// {
+//     {
+//         f()
+//         sstore(0, 1)
+//         f()
+//         sstore(0, 1)
+//         f()
+//         sstore(0, 1)
+//     }
+//     function f()
+//     {
+//         let b := 10
+//         let _1 := 0
+//         let a := calldataload(_1)
+//         for { } iszero(b) { b := add(b, not(0)) }
+//         { mstore(a, _1) }
+//     }
```

Comment:
I'm a bit surprised by how it was able to infer `a` was unused.

Reply:
Don't think it is a bug, since this is how the same test runs for `UnusedFunctionParameterPruner`.
```solidity
{
    sstore(f(1), 1)
    sstore(f(2), 1)
    sstore(f(3), 1)
    function f(a) -> x {
        let b := 10
        for {} iszero(b) { b := sub(b, 1) }
        {
            a := calldataload(0)
            mstore(a, x)
        }
    }
}
// ----
// step: unusedFunctionParameterPruner
//
// {
//     sstore(f_1(1), 1)
//     sstore(f_1(2), 1)
//     sstore(f_1(3), 1)
//     function f(a)
//     {
//         let b := 10
//         for { } iszero(b) { b := sub(b, 1) }
//         {
//             a := calldataload(0)
//             mstore(a, 0)
//         }
//     }
//     function f_1(a_2) -> x_3
//     { f(a_2) }
// }
```

Reply:
I think what happens is SSATransform (a) + UnusedPruner (u) + RedundantAssignEliminator (r) is able to remove `a` from the function. And VarNameCleaner cleans up some `a_n` back to `a`. To "verify" the claim, we can replace the parameter name `a` by `a_5`. And still get the same result.